enhancements/storage/csi-driver-global-configmap-secret-mounter.md

```diff
@@ -0,0 +1,201 @@
+---
+title: global-configmap-secret-mounter
+authors:
+  - "@janedoe"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Global ConfigMap and Secret Mounter
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+ 1.  It is possible to build some sorts of per-namespace ACL boundaries if we really wanted to, but I think it makes more sense
+ to have the idea of global or namespaced and no in-between since nothing else is in between today.  They can write a controller
+ if they really want to create resources shared across some namespaces and not others.
+
+## Summary
+
+A CSI driver that can mount a configmap or secret from a specified namespace into any pod.
```

Comment:
Wouldn't this work better with rbac and GC if there were a non-namespaced ConfigMap and non-namespaced Secret object, instead of allowing what amounts to cross-namespace mounting of secrets?

Reply:
> Wouldn't this work better with rbac and GC if there were a non-namespaced ConfigMap and non-namespaced Secret object, instead of allowing what amounts to cross-namespace mounting of secrets?

I would rather build on existing concepts in usages are that are inherently low cardinality to reflect what people want to use these for.  This is equivalent of exposure we desired from our high privilege namespaces like openshift-config and openshift-config-managed.